storage/range_command.go

```diff
@@ -961,7 +961,7 @@ func (r *Range) AdminMerge(args *proto.AdminMergeRequest, reply *proto.AdminMerg
 		// Noop.
 		return
 	}
-	subsumedRng := r.rm.LookupRange(desc.EndKey, desc.EndKey)
+	subsumedRng := r.rm.LookupRange(desc.EndKey, desc.EndKey.Next())
```

Comment:
Instead of adding `.Next()` everywhere we do LookupRange, let's make it support nil as the end key.  You'll need to adjust `storage/engine/keys.go:KeyAddress()` to return nil if nil is supplied.